gr-pdu/lib/pdu_set_impl.cc

```diff
@@ -29,10 +29,9 @@ pdu_set_impl::pdu_set_impl(pmt::pmt_t k, pmt::pmt_t v)
       d_k(k),
       d_v(v)
 {
-    message_port_register_out(pdu::pdu_port_id());
-    message_port_register_in(pdu::pdu_port_id());
-    set_msg_handler(pdu::pdu_port_id(),
-                    [this](pmt::pmt_t msg) { this->handle_msg(msg); });
+    message_port_register_out(ports::pdus());
+    message_port_register_in(ports::pdus());
+    set_msg_handler(ports::pdus(), [this](pmt::pmt_t msg) { this->handle_msg(msg); });
```

Comment:
Correct me, but if we change this lambda's, and the msg handler's, signature to `const pmt::pmt_t& msg`, we'd be saving the schmart pointer a ref increase (and decrease), which might be nice, seeing that these `shared_ptr`s imply refcount change atomicity.

Note that this is a general architectural problem with PMTs, not limited to this block, at all, but, knowing that you build high-rate applications out of this, this might be a good place to start that practice.

Definitely not a blocker here, but `perf` / kernelshark can be pretty harsh on how often we change refcounts.

Reply:
I think this makes sense in general but should be a separate issue/PR to cover everything in-tree IMO.